### DISCUSSION TITLE
[Feature Request] Table component : fix columns

### DISCUSSION THREAD

**Discussion by AntoineArt** [heroui-inc/heroui#1281] [Feature requests]
### Is your feature request related to a problem? Please describe.

For large tables, it is important to be able to fix one or several columns. 
For example if I have a table listing users with 15 info columns, i need to be able to see the user's name even if I'm looking at the last columns.

### Describe the solution you'd like

I would like to be able to fix one or several columns, either left or right, with a simple prop. 
For example "fixLeftColumns=2" would fix the first two left columns.

I would like to be able to easily see that a column or set of columns is fixed. 
For example if "fixLeftColumns=2", then the second left column could have a shadow on its right side.
Or any other aesthetic way to show that, with the quality next UI is known for.

### Describe alternatives you've considered

You have an example of the base feature here : https://react-table-library.com/?path=/docs/compact-table--fixedcolumn

react-table-library is fine but doesn't have the finesse and the aesthetic next UI has. Plus it would be amazing to not have to use many UI libraries for a single project.

**Comment by vaynevayne** [comment]
aria-table not support this feature,  i think  need aria's member to do this, 
It's not just fixed columns, there are some other features that are missing, such as ( row-drag column-drag column-resize(source code has includes this, but need to complete), float-scrollbar) and I think it's a good idea to  use [react-table](https://tanstack.com/table/v8) for integration.

**Comment by thunder6230** [comment]
We are working on tha tables right now and it's really desireable to have at least the sticky columns feature built in

**Comment by DophinL** [comment]
Really need this feature !

**Comment by vietvh-3042** [comment]
+1

**Comment by matheusfscit** [comment]
+1

**Comment by BorisBarzotto** [comment]
This feature would be very much appreciated, it would be a very good plus for the table. Cheers

**Comment by SainDev** [comment]
+1

**Comment by jimapl** [comment]
+2

**Comment by AvuxDemons** [comment]
+1

**Comment by scott-koalatech** [comment]
+1

**Comment by revoyrm** [comment]
+1

**Comment by gilbert-hirint** [comment]
+1

**Comment by VladimirTomko** [comment]
+1

**Comment by Jhonatanzr** [comment]
+1

**Comment by abdielgzzmtz** [comment]
+1

**Comment by 201549BL** [comment]
+1

**Comment by dannielirvin** [comment]
+1
